include/deal.II/lac/block_linear_operator.h

```diff
@@ -662,6 +662,177 @@ block_triangular_inverse(const BlockMatrix &block_matrix,
   return result * inverse_diagonal;
 }
 
+/**
+ * @relates LinearOperator
+ *
+ * This function implement a gauss elimination argument to invert a lower
+ * block triangular matrix.
```

Comment:
I believe this process is mostly known as Backward substitution (or Forward), i.e.,

https://en.wikipedia.org/wiki/Triangular_matrix#Forward_and_back_substitution

May be rename it and insert a Mathematical description of the algorithm as in the link above, adapted for matrices?